src/operator/tensor/cast_storage-inl.cuh

```diff
@@ -188,6 +188,15 @@ struct FillRspValsKernel {
   }
 };
 
+template<typename xpu, int ndim, typename DType>
+inline mshadow::Tensor<xpu, ndim, DType> AllocateTempDataForCast(const OpContext& op_ctx,
```

Comment:
2) It is not gpu-specific, and we may find it necessary for CPU casts as well, in which case it can be moved.  That being said, it is a temporary hack and as the first item stated, I am trying to limit its use to this file and cast only.  Putting it in some general place with a general name will cause people to use it for general-purpose temp allocation.